solt/transforms/_transforms.py

```diff
@@ -1122,17 +1137,19 @@ def sample_transform(self, data):
             kernel = kernel / np.sum(kernel)
             self.state_dict.update({"motion_kernel": kernel})
 
-    @img_shape_checker
     def _apply_img(self, img: np.ndarray, settings: dict):
+        ensure_valid_image(img, num_dims=(2,))
+        shape_in = img.shape
+
         if self.blur == "g":
             return cv2.GaussianBlur(
                 img, ksize=(self.state_dict["k_size"], self.state_dict["k_size"]), sigmaX=self.state_dict["sigma"],
-            )
+            ).reshape(shape_in)
```

Comment:
This reshape is not needed here given you will do the needed reshaping before returning the result.

Reply:
See the comment above. It is needed, since OpenCV is corrupting the shape for the single-channel images.

Reply:
I think adds more complexity to the codestyle and it diminishes the ease of extendibility of SOLT.